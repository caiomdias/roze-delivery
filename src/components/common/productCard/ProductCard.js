import React, { useState } from 'react';

const ProductCard = ({ mode, product }) => {
	const [ quantity, setQuantity ] = useState(0);
	const [ productState, setProductState ] = useState(product)
	
	const updateQuantity = (type) => {
		if (type === 'add') {
			setQuantity(quantity + 1);
		} else {
			if (quantity <= 1) return quantity;
			setQuantity(quantity - 1);
		}
	};

	return (
		<div className={`product-card product-card--${mode}`}>
			<div className="product-card__content">
				<div className="product-card__image">
					<img src={productState.images && productState.images[0].url} alt={`Imagem de ${productState.title}`}/>
				</div>
				<div className="product-card__description">
					<h4 className="product-card__title">{productState.title}</h4>
					<p className="product-card__price">R$ {productState.productVariants && productState.productVariants[0].price}</p>
				</div>
			</div>
			<div className="product-card__actions">
				<div className="quantity">
					<div
						onClick={() => updateQuantity('remove')}
						className="quantity__minus">-</div>
					<div className="quantity__value">{quantity}</div>
					<div
						onClick={() => updateQuantity('add')}
						className="quantity__plus">+</div>
				</div>
				{
					mode === 'shelf' && (
						<div className="buy-button">
							Adicionar
						</div>
					)
				}
				{
					mode === 'cart' && (
						<a href="">Excluir</a>
					)
				}
			</div>
		</div>)
};

ProductCard.defaultProps = {
	mode: 'shelf',
	product: {}
};

export default ProductCard;
