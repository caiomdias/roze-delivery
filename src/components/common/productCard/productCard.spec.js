import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard from './ProductCard';

describe("Product Cart Component", () => {
	it('render correctly ProductCard component', () => {  
		const productCardComponent = renderer.create(<ProductCard />);
		expect(productCardComponent).toMatchSnapshot();
	});
});