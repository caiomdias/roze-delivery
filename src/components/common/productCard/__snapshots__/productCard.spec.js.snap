// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Cart Component render correctly ProductCard component 1`] = `
<div
  className="product-card product-card--shelf"
>
  <div
    className="product-card__content"
  >
    <div
      className="product-card__image"
    >
      <img
        alt="Imagem de undefined"
      />
    </div>
    <div
      className="product-card__description"
    >
      <h4
        className="product-card__title"
      />
      <p
        className="product-card__price"
      >
        R$ 
      </p>
    </div>
  </div>
  <div
    className="product-card__actions"
  >
    <div
      className="quantity"
    >
      <div
        className="quantity__minus"
        onClick={[Function]}
      >
        -
      </div>
      <div
        className="quantity__value"
      >
        0
      </div>
      <div
        className="quantity__plus"
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      className="buy-button"
    >
      Adicionar
    </div>
  </div>
</div>
`;
